src/VirtoCommerce.AzureSearchModule.Data/AzureSearchProvider.cs

```diff
@@ -182,9 +182,22 @@ protected virtual SearchDocument ConvertToProviderDocument(IndexDocument documen
                     var isCollection = providerField.Type.ToString().StartsWith("Collection(");
 
                     var point = field.Value as GeoPoint;
-                    var value = point != null
-                        ? (isCollection ? field.Values.Select(v => ((GeoPoint)v).ToDocumentValue()).ToArray() : point.ToDocumentValue())
-                        : (isCollection ? field.Values : field.Value);
+                    object value;
+                    if (point != null)
+                    {
+                        if (isCollection)
+                            value = field.Values.Select(v => ((GeoPoint)v).ToDocumentValue()).ToArray();
+                        else
+                            value = point.ToDocumentValue();
+                    }
+                    else
+                    {
+                        if (isCollection)
+                            value = field.Values;
+                        else
+                            value = field.Value;
+
+                    }
```

Comment:
https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-8#tuple-patterns
``` csharp
object value = (field.Value, isCollection) switch =>
{
    (GeoPoint, true) => field.Values.Select(v => ((GeoPoint)v).ToDocumentValue()).ToArray(),
    (GeoPoint point, false) => point.ToDocumentValue(),
    (_, true) => field.Values,
    (_, false) => field.Value
};
```

Reply:
Can you use something like that?

Reply:
revert back